src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -440,6 +522,26 @@ private class AssemblyMetadata : PackageMetadata
             }
         }
 
+        /// <summary>
+        /// Represents metadata for a Reference item that we want to pretend was resolved as a
+        /// standard NuGet package assembly.
+        /// </summary>
+        private class PreResolvedNetStandardAssemblyMetadata : PackageMetadata
+        {
+            public PreResolvedNetStandardAssemblyMetadata(string name, string path)
+                : base(
+                      type: DependencyType.Assembly,
+                      dependencies: null,
+                      isTopLevelDependency: false,
+                      name: name,
+                      version: string.Empty,
+                      path: path,
+                      resolved: true,
+                      isImplicitlyDefined: false)
```

Comment:
There is code that was added in VS to see if an item is defined in the project and if not then make it unremovable. So not sure if this is still used.